services/dynamodb/src/test/java/software/amazon/awssdk/services/dynamodb/DynamoDbRetryPolicyTest.java

```diff
@@ -55,4 +59,32 @@ public void test_backoffBaseDelay_with_standardRetryPolicy() {
         });
     }
 
+    @Test
+    public void resolve_retryModeSetInEnv_doesNotCallSupplier() {
```

Comment:
What about the cases where you 
- supply both an env var and profile file
- none (not sure if this is relevant)

Reply:
Added three test cases:

- Env var and profile file
- Sdk client option
- none

Reply:
Nice!